src/Cameras/Inputs/freeCameraMouseWheelInput.ts

```diff
@@ -0,0 +1,402 @@
+import { Nullable } from "../../types";
+import { serialize } from "../../Misc/decorators";
+import { FreeCamera } from "../../Cameras/freeCamera";
+import { CameraInputTypes } from "../../Cameras/cameraInputsManager";
+import { BaseCameraMouseWheelInput } from "../../Cameras/Inputs/BaseCameraMouseWheelInput";
+import { Matrix, Vector3 } from "../../Maths/math.vector";
+
+/**
+ * Defines the potential axis to be altered in a transform operation.
+ */
+export enum AXIS {
+    /**
+     * X axis.
+     */
+    X,
+    /**
+     * Y axis.
+     */
+    Y,
+    /**
+     * Z axis.
+     */
+    Z
+}
+
+/**
+ * An interface for user definable callback to be called on mouse wheel movement.
+ */
+export interface IFreeCameraMouseWheelCustomCallback {
```

Comment:
I just discovered Babylon's Observer pattern.
This section would be better implemented using that.